spark/src/main/scala/org/apache/spark/sql/delta/Checksum.scala

```diff
@@ -795,8 +795,11 @@ trait ValidateChecksum extends DeltaLogging { self: Snapshot =>
       }
     }
 
-    compareAction(checksum.metadata, computedStateToCheckAgainst.metadata, "Metadata", "metadata")
-    compareAction(checksum.protocol, computedStateToCheckAgainst.protocol, "Protocol", "protocol")
+    if (spark.sessionState.conf.getConf(
+        DeltaSQLConf.USE_PROTOCOL_AND_METADATA_FROM_CHECKSUM_ENABLED)) {
```

Comment:
Is it useful/important to make this check conditional, where it was previously unconditional?
What is the rationale?

Reply:
I though about this again, and this does not make much sense. Removed